pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -30,13 +33,73 @@
 import org.apache.pinot.common.request.Literal;
 import org.apache.pinot.common.request.PinotQuery;
 import org.apache.pinot.spi.utils.BytesUtils;
+import org.apache.pinot.spi.utils.CommonConstants;
 import org.apache.pinot.sql.FilterKind;
+import org.apache.pinot.sql.parsers.CalciteSqlParser;
+import org.apache.pinot.sql.parsers.SqlCompilationException;
+import org.apache.pinot.sql.parsers.SqlNodeAndOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 public class RequestUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RequestUtils.class);
+
   private RequestUtils() {
   }
 
+  public static SqlNodeAndOptions parseQuery(long requestId, String query, JsonNode request)
```

Comment:
Let's not pass in `requestId` since it is specific to the broker, and only for logging purpose. If necessary, we can log the options in the broker request handler.